netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -194,6 +203,33 @@ public ClientStream newStream(
     if (channel == null) {
       return new FailingClientStream(statusExplainingWhyTheChannelIsNull, tracers);
     }
+    if (negotiator instanceof ClientTlsProtocolNegotiator && callOptions.getAuthority() != null) {
```

Comment:
This implementation should be moved to the ClientTlsProtocolNegotiator itself, and we add a negotiator API to call here.

Reply:
Done.